dot/rpc/modules/chain_test.go

```diff
@@ -15,3 +15,473 @@
 // along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
 
 package modules
+
+import (
+	"errors"
+	"math/big"
+	"net/http"
+	"testing"
+
+	apimocks "github.com/ChainSafe/gossamer/dot/rpc/modules/mocks"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChainModule_GetBlock(t *testing.T) {
+	mockedHash := common.NewHash([]byte{0x01, 0x02})
+	emptyBlock := types.NewEmptyBlock()
+
+	mockBlockAPI := new(apimocks.BlockAPI)
+	mockBlockAPI.On("GetBlockByHash", mock.AnythingOfType("common.Hash")).Return(&emptyBlock, nil)
```

Comment:
Try to avoid `mock.AnythingOfType` / `mock.Anything`, instead give it what it should expect. I guess it should be deterministic here. If it has some randomness factor, then you need to mock the randomness so it's always the same in the test.

I personally use `mock.AnythingOfType` when for example it's a timing out context, which you can't predict or mock. But it's rather rare.